### PR TITLE
Check integers in both parameters 1

### PR DESCRIPTION
**Pull request by JawidMohseni**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

**Comment by JawidMohseni** [comment]
Hi everyone! This is my first attempt of the pull request. Could someone please review my code?

**Comment by asagulenko** [comment]
Jawid, I'm working on it right now!

File: solutions/only_ints.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+A module for verifying integer inputs.
+
+Module contents:
+    - only_ints: A function to check if two parameters are both integers.
+
+Created on 12 29 2024
```
Comment (asagulenko) [inline solutions/only_ints.py:9]: Maybe put data in some format like yyyy.mm.dd or yyyy/mm/dd?

File: solutions/tests/test_only_ints.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    def test_both_integers(self):
+        """Test case where both parameters are integers."""
+        self.assertTrue(only_ints(1, 2))  # Should return True
+
+    def test_first_is_string(self):
+        """Test case where the first parameter is not an integer"""
+        self.assertFalse(only_ints("a", 1))  # Should return False
+
+    def test_both_integers_minus(self):
+        """Test case where both parameters are integers."""
```
Comment (asagulenko) [inline solutions/tests/test_only_ints.py:32]: description of the third test is exactly the same as the first test case. Please edit

File: solutions/only_ints.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+@author: Mohammad
```
Comment (asagulenko) [inline solutions/only_ints.py:10]: Doctests are missing (at least three passing doctests)

File: solutions/only_ints.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    param1 (int): The first parameter to check, expected to be an integer.
+    param2 (int): The second parameter to check, expected to be an integer.
+
+    Returns:
+    bool: True if both parameters are integers, False otherwise.
+
+    Raises:
+    TypeError: If any of the parameters is not an integer.
+    """
+    if not isinstance(param1, int) or not isinstance(param2, int):
```
Comment (asagulenko) [inline solutions/only_ints.py:29]: what about putting "positive" if case? like 
if isinstance(param1, int) and isinstance(param2, int): #I may be mistaken with syntax

**Review by asagulenko — CHANGES_REQUESTED** [review]
Please see the comments for the code

File: solutions/only_ints.py (hunk 1 above)
Comment (JawidMohseni) [reply solutions/only_ints.py:9]: Agree , I will do it

File: solutions/only_ints.py (hunk 4 above)
Comment (JawidMohseni) [reply solutions/only_ints.py:29]: Oh I almost forgot about it, I can make it make it more clear

File: solutions/only_ints.py (hunk 3 above)
Comment (JawidMohseni) [reply solutions/only_ints.py:10]: I almost forgot about it—this can be added to the code for better functionality.

File: solutions/tests/test_only_ints.py (hunk 2 above)
Comment (JawidMohseni) [reply solutions/tests/test_only_ints.py:32]: Thank you for reminding it

File: solutions/only_ints.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    Parameters:
+    param1 (int): The first parameter to check, expected to be an integer.
+    param2 (int): The second parameter to check, expected to be an integer.
+
+    Returns:
+    bool: True if both parameters are integers, False otherwise.
+
+    Raises:
+    TypeError: If any of the parameters is not an integer.
```
Comment (ggmeklit) [inline solutions/only_ints.py:27]: I dont think this sentence is correct. If any of the parameter are not integer it turns false not typeerror.

**Review by ggmeklit — CHANGES_REQUESTED** [review]
- in addition to the docstring and Anna's feedback  I dont believe it raises a TypeError as stated in your docstring. Your file handles any case by Turning True or False. You may test if this is wrong.
- I see that you have test for when one is a string, perhaps also consider float as well (this is just an idea)
-Overall your code is clearly written and very easy to follow

File: solutions/only_ints.py (hunk 5 above)
Comment (JawidMohseni) [reply solutions/only_ints.py:27]: How about if the change the if statement to something like this 

if not isinstance(param1, int) or not isinstance(param2, int):
        raise TypeError("Both parameters must be integers.")  # Raise TypeError

    return True
    I just tests it and if any of the parma is not integers , it will be a Raise TypeError with a meassge that says " TypeError: Both parameters must be integers."

Comment (JawidMohseni) [reply solutions/only_ints.py:27]: Hi Meklit,

I wanted to update you on the testing for the only_ints function again, While working on the test cases, I noticed that the Raises clause in the docstring was causing some confusion. Specifically, the function does not raise a TypeError as stated—it simply returns False for non-integer inputs.

After reviewing the implementation and testing, I removed the Raises section from the docstring to ensure it accurately reflects the function's behavior. This should help avoid further confusion moving forward.

If there is no review from you guys so I will finish it and push back with the updated version 

Thank you

**Review by asagulenko — APPROVED** [review]

**Review by ggmeklit — APPROVED** [review]
